From 3bbf8a159ece1d941ef7249309dedd463897a4ad Mon Sep 17 00:00:00 2001
From: Samuel Pitoiset <samuel.pitoiset@gmail.com>
Date: Wed, 11 Dec 2024 11:39:52 +0100
Subject: [PATCH 5/6] radv: fix disabling DCC for stores with drirc

Displayable DCC should also be disabled, otherwise it's asserting
somewhere in ac_surface.c

Fixes: e3d1f27b318 ("radv: add radv_disable_dcc_stores and enable for Indiana Jones: The Great Circle")
Signed-off-by: Samuel Pitoiset <samuel.pitoiset@gmail.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/32584>
(cherry picked from commit 4d1aa9a2d0858c975e365a63234301256fa9cc77)
---
 src/amd/vulkan/radv_formats.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/amd/vulkan/radv_formats.c b/src/amd/vulkan/radv_formats.c
index b56cf52f275..d8410814553 100644
--- a/src/amd/vulkan/radv_formats.c
+++ b/src/amd/vulkan/radv_formats.c
@@ -627,10 +627,10 @@ radv_get_modifier_flags(struct radv_physical_device *pdev, VkFormat format, uint
          return 0;
 
       /* Only disable support for STORAGE_IMAGE on modifiers that
-       * do not support DCC image stores.
+       * do not support DCC image stores or when explicitly disabled.
        */
       if (!ac_modifier_supports_dcc_image_stores(pdev->info.gfx_level, modifier) ||
-          radv_is_atomic_format_supported(format))
+          radv_is_atomic_format_supported(format) || instance->drirc.disable_dcc_stores)
          features &= ~VK_FORMAT_FEATURE_2_STORAGE_IMAGE_BIT;
 
       if (instance->debug_flags & (RADV_DEBUG_NO_DCC | RADV_DEBUG_NO_DISPLAY_DCC))
-- 
2.47.1

